src/modules/fw_att_control/FixedwingAttitudeControl.cpp

```diff
@@ -325,16 +324,22 @@ void FixedwingAttitudeControl::Run()
 			/* Run attitude controllers */
 
 			if (_vcontrol_mode.flag_control_attitude_enabled && _in_fw_or_transition_wo_tailsitter_transition) {
-				if (PX4_ISFINITE(_att_sp.roll_body) && PX4_ISFINITE(_att_sp.pitch_body)) {
-					_roll_ctrl.control_roll(_att_sp.roll_body, _yaw_ctrl.get_euler_rate_setpoint(), euler_angles.phi(),
+				Eulerf setpoint(Quatf(_att_sp.q_d));
+				float roll_body = setpoint.phi();
+				float pitch_body = setpoint.theta();
+
+				if (PX4_ISFINITE(roll_body) && PX4_ISFINITE(pitch_body)) {
```

Comment:
Is it even possible that one euler angle is infinite while the other is not?

Reply:
I don't think it is ever possible to have an infinite value from a quaternion :) It is also unclear what the controller is doing if it is invalid. I will remove the condition if this is fine for you.